cmd/src/validate.go

```diff
@@ -0,0 +1,609 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strings"
+	"text/template"
+	"time"
+
+	"github.com/ghodss/yaml"
+	jsoniter "github.com/json-iterator/go"
+	"github.com/mattn/go-isatty"
+)
+
+type validationSpec struct {
+	FirstAdmin struct {
+		Email    string
+		Username string
+		Password string
+	}
+	WaitRepoCloned struct {
+		Repo                     string
+		MaxTries                 int
+		SleepBetweenTriesSeconds int
+	}
+	SearchQuery     string
+	ExternalService struct {
+		Kind           string
+		DisplayName    string
+		Config         *json.RawMessage
+		DeleteWhenDone bool
+	}
+}
+
+type validator struct {
+	client *vdClient
+}
+
+func init() {
+	usage := `'src validate' is a tool that validates a Sourcegraph instance.
+
+EXPERIMENTAL: 'validate' is an experimental command in the 'src' tool.
+
+Usage:
+
+	src validate [options] <script-file>
+or
+    cat <script-file> | src validate [options]
+`
+	flagSet := flag.NewFlagSet("validate", flag.ExitOnError)
+	usageFunc := func() {
+		_, _ = fmt.Fprintf(flag.CommandLine.Output(), "Usage of 'src validate %s':\n", flagSet.Name())
+		flagSet.PrintDefaults()
+		fmt.Println(usage)
+	}
+
+	var (
+		contextFlag = flagSet.String("context", "", `Comma-separated list of key=value pairs to add to the script execution context`)
+		docFlag     = flagSet.Bool("doc", false, `Show documentation`)
```

Comment:
Remove and just have it print the URL as part of the help output instead? I don't understand why a `-doc` flag is useful when `-h` exists for this purpose :)